blog/17-04-2023-implementing-multi-factor-authentication-on-celos-blockchain-using-smart-contracts/implementing-multi-factor-authentication-on-celos-blockchain-using-smart-contracts.md

```diff
@@ -0,0 +1,321 @@
+
+`getSecret`: This function prompts the user to enter the account address, and then calls the "getSecret" function on the MFA contract with the given parameter. It logs the secret associated with the given account address.
+
+`checkCode`: This function prompts the user to enter the account address and code, and then calls the "checkCode" function on the MFA contract with the given parameters. It logs a message indicating whether the code is valid for the given account address.
+
+All three functions use the web3 library to interact with the MFA contract deployed on the Celo network. The Web3 library is initialized with a URL for a Celo network node, which is used to connect to the network and send transactions.
+
+In the `main` function, we prompt the user for the action they want to perform and call the corresponding function. If an invalid action is entered, we display an error message.
+
+Finally, we can run the mfa-cli.js file in the terminal using the following command:
+
+```bash
+node mfa-cli.js
+```
+
+This will prompt the user for an action to perform and then perform the corresponding action based on the user's input.
+
+Congratulations! You have successfully created a simple MFA system using Celo and Node.js. With this basic system in place, you can now build on it to add more functionality, such as allowing users to enable and disable MFA for their accounts or setting up backup codes for users to use if they lose access to their primary MFA device.
+
+## Conclusion​
+
+In this tutorial, we've shown how to create a simple MFA system using the Celo blockchain and javascript. We started by creating a smart contract to store MFA secrets and validate codes; then we used the web3 library to interact with the contract from a javascript application. Finally, we built a simple command-line interface to allow users to set, retrieve, and validate MFA secrets.
+
+This is just a starting point, and there are many ways to extend and improve this system.
+
+We hope this tutorial has helped get you started with building MFA systems on the Celo blockchain.
+
+## Next Steps​
+
+There are many ways to improve the MFA contract and the CLI tool, such as adding more authentication factors and making the contract more secure. Here are some suggestions for the next steps:
+
+- Add error handling and validation for user input to make the CLI tool more robust.
+
+- Create a web-based interface for the MFA tool using a framework like React or Angular.
+
```

Comment:
Although this is suggestive can you add links to other tutorials or videos to help the users navigate to the next step, preferably celo articles, or any other piece

Reply:
okay